src/modules/75-cf/pages/onboarding/views/CreateAFlagView.tsx

```diff
@@ -5,79 +5,150 @@
-import css from './CreateAFlagView.module.scss'
-
+import { OnboardingSelectedFlag } from '../OnboardingSelectedFlag'
 export interface CreateAFlagViewProps {
-  setFlagName: React.Dispatch<React.SetStateAction<string>>
-  flagInfo: FeatureFlagRequestRequestBody
-  isCreated: boolean
-  goNext: () => void
+  selectedFlag?: Feature
+  setSelectedFlag: (flag?: Feature) => void
 }
 
-export const CreateAFlagView: React.FC<CreateAFlagViewProps> = ({ flagInfo, setFlagName, isCreated, goNext }) => {
+export const CreateAFlagView: React.FC<CreateAFlagViewProps> = ({ selectedFlag, setSelectedFlag }) => {
+  const { showError } = useToaster()
   const { getString } = useStrings()
-
   const { trackEvent } = useTelemetry()
+  const [searchTerm, setSearchTerm] = useState<string>('')
+  const [flagCreated, setFlagCreated] = useState(false)
+  const { orgIdentifier, accountId: accountIdentifier, projectIdentifier } = useParams<Record<string, string>>()
+
+  const { mutate: createFeatureFlag, loading: isLoadingCreateFeatureFlag } = useCreateFeatureFlag({
+    queryParams: {
+      accountIdentifier,
+      orgIdentifier
+    } as CreateFeatureFlagQueryParams
+  })
+
+  const queryParams = {
+    projectIdentifier,
+    accountIdentifier,
+    orgIdentifier,
+    name: searchTerm,
+    pageSize: CF_DEFAULT_PAGE_SIZE
+  }
+
+  const { data: allFeatureFlags, refetch: fetchFlags } = useGetAllFeatures({
+    lazy: true,
+    queryParams,
+    debounce: 250
+  })
 
   useEffect(() => {
     trackEvent(FeatureActions.CreateAFlagView, {
       category: Category.FEATUREFLAG
     })
+    fetchFlags()
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [])
 
+  useEffect(() => {
+    fetchFlags()
+  }, [searchTerm, fetchFlags])
+
+  const createNewFlag = (flagName: string): void => {
+    const flagData: FeatureFlagRequestRequestBody = {
+      project: projectIdentifier,
+      name: flagName,
+      identifier: flagName.toLowerCase().trim().replace(/\s|-/g, '_'),
+      kind: 'boolean',
+      archived: false,
+      variations: [
+        { identifier: 'true', name: 'True', value: 'true' },
+        { identifier: 'false', name: 'False', value: 'false' }
+      ],
+      defaultOnVariation: 'true',
+      defaultOffVariation: 'false',
+      permanent: false
+    }
+
+    createFeatureFlag(flagData)
+      .then(resp => {
+        const flag = resp?.details?.governanceMetadata?.input.flag
+        setSelectedFlag(flag)
+        setFlagCreated(true)
+        fetchFlags()
+      })
+      .catch(error => showError(getErrorMessage(error), undefined, 'cf.create.ff.error'))
+  }
+
+  const onChangeSelect = (selectedOption: string): void => {
+    if (!allFeatureFlags?.features?.some(feat => feat.identifier === selectedOption)) {
+      createNewFlag(selectedOption)
+    } else {
+      const flag = allFeatureFlags.features.find(feat => feat.identifier === selectedOption)
+      setSelectedFlag(flag)
+    }
```

Comment:
You might as well only search `features` once
```suggestion
    const flag = allFeatureFlags?.features?.find(feat => feat.identifier === selectedOption)
    if (!flag) {
      createNewFlag(selectedOption)
    } else {
      setSelectedFlag(flag)
    }
```